### DISCUSSION TITLE
warning: Using the last argument as keyword parameters is deprecated

### DISCUSSION THREAD

**Discussion by cseas** [mmistakes/minimal-mistakes#2764] [Q&A] [answered]
Trying to run the theme lately repeatedly throws the following warning:

```
/home/abhijeet/gems/gems/pathutil-0.16.2/lib/pathutil.rb:502: warning: Using the last argument as keyword parameters is deprecated
```

This issue is well known and exists in Jekyll 3.x 
https://github.com/jekyll/jekyll/issues/7947
A simple way to make the warning go away is to upgrade Jekyll to 4.x
But it seems this theme is dependant on the github-pages gem which has been stuck on Jekyll 3.x since a long time
https://github.com/github/pages-gem/issues/651

So I understand that there is no simple way to make the warning go away by upgrading to Jekyll 4 if we want to use the github-pages gem. But since github-pages isn't showing great support, is there a way to remove that dependency altogether? Then we might be able to update to Jekyll 4. That would require manually setting up some CI to publish to `gh-pages` branch but that's fine.

## Environment

- Minimal Mistakes version: Using remote theme
- Ruby gem or remote theme version: How do I find out?
- Jekyll version: Stuck at 3.7
- Git repository URL: https://github.com/cseas/cseas.github.io
- GitHub Pages hosted (if yes provide URL to site): www.absingh.com
- Operating system: Ubuntu 20.04 LTS

## Expected behavior

We should be able to update to Jekyll 4+ to avoid the deprecation warnings.

## Steps to reproduce the behavior

`bundle exec jekyll serve`

**Comment by cseas** [comment]
Why this is not #2458:
The discussion in that Issue suggests that upgrading to Jekyll 4.1 will solve the issue.
But that's exactly what I'm trying to do. The `github-pages` dependency is stuck at Jekyll 3.x and stops the upgrade.

So it would be great if there were some steps in the documentation to explain how to remove that dependency and use this theme with Jekyll 4.

**Comment by mmistakes** [comment, marked as answer]
If you want to use 4.1, replace `gitHub-pages` with `jekyll`, run `bundle update`.

Then you’ll need to look into using a GitHub workflow/action to build your site and push it up to GH Pages since you won’t be using the GH Pages gem anymore. 

https://jekyllrb.com/docs/continuous-integration/github-actions/
